packages/react/src/components/loaded/loaded.test.tsx

```diff
@@ -0,0 +1,92 @@
+import { render, screen } from "@testing-library/react";
+import { SlashIDLoaded } from ".";
+import { TestSlashIDProvider } from "../../context/test-slash-id-provider";
+import { SDKState } from "../../context/slash-id-context";
+import { faker } from '@faker-js/faker'
+import { sdkNotReadyStates } from "../test-utils";
+
+const Mock = ({ text }: { text: string }) => (
+  <h1>
+    {text}
+  </h1>
+);
+
+describe("SlashIDLoaded", () => {
+  for (const state of sdkNotReadyStates) {
+    test(`should not render children when sdk state is '${state}'`, () => {
+      const text = faker.lorem.sentence()
+
+      render(
+        <TestSlashIDProvider sdkState={state}>
+          <SlashIDLoaded>
+            <Mock text={text} />
+          </SlashIDLoaded>
+        </TestSlashIDProvider>
+      );
+      expect(screen.queryByText(text)).not.toBeInTheDocument();
+    });
+  }
+
+  for (const state of sdkNotReadyStates) {
+    test(`should render children when sdk state is '${state}'`, () => {
+      const text = faker.lorem.sentence()
+
+      render(
+        <TestSlashIDProvider sdkState={state}>
+          <SlashIDLoaded>
+            <Mock text={text} />
+          </SlashIDLoaded>
+        </TestSlashIDProvider>
+      );
+      expect(screen.queryByText(text)).not.toBeInTheDocument();
+    });
+  }
+
+  for (const state of sdkNotReadyStates) {
+    test(`should render fallback when sdk state is '${state}'`, () => {
+      const text = faker.lorem.sentence()
+      const fallback = faker.lorem.sentence()
+
+      render(
+        <TestSlashIDProvider sdkState={state}>
+          <SlashIDLoaded
+            fallback={<Mock text={fallback} />}
```

Comment:
No not at all, passing JSX to a prop is standard and encouraged. This is one of the key strengths of JSX in general compared to other templating languages.

The React builtin component `<Suspense>` does exactly this:
https://react.dev/reference/react/Suspense

What you're suggesting isn't necessarily bad but I feel like the trade off might not be worth it because it increases our API surface, whereas this pattern is well understood by React developers and it keeps our API surface smaller.